src/encoder.rs

```diff
@@ -347,6 +351,61 @@ impl<T: Pixel> FrameState<T> {
     }
   }
 
+  /// Computes the variance of 8x8 blocks of the luma plane and stores twice the
+  /// SD in a decimated (by 8 on both axes) activity plane.
+  ///
+  /// # Panics
+  ///
+  /// Panics if the activity mask is not decimated by 8 on both axes.
+  ///
+  /// Note : The plane is also normalized to a 16-bit scale.
+  pub fn compute_activity_mask(&mut self) {
+    
+    let luma_plane = &self.input.planes[0];
+    let PlaneConfig { width, height, .. } = luma_plane.cfg;
+    let luma = &luma_plane.data;
+    let PlaneConfig { xdec, ydec, .. } = self.activity_mask.cfg;
+    assert!(xdec == 3);
+    assert!(ydec == 3);
+    let width = (width >> xdec) << xdec;
+    let height = (height >> ydec) << ydec;
+    let mut mean = vec![0u32; (width >> xdec) * (height>>ydec)];
+    let mut var = vec![0u32; (width >> xdec) * (height>>ydec)];
+
+    //Computing the mean of each 8x8 block
+    for r in 0..height {
+      for c in 0..width {
+        let luma_offset = luma_plane.index(c, r);
+        let act_offset = (r>>ydec) * (width>>ydec) + (c>>xdec);
+        let temp_luma: u32 = luma[luma_offset].into(); 
+        mean[act_offset] += temp_luma;
+      }
+    }
+    mean.iter_mut().for_each(|sum| *sum >>= xdec+ydec);
+
+    //Computing the variance of each 8x8 block
+    for r in 0..height {
+      for c in 0..width {
+        let luma_offset = luma_plane.index(c, r);
+        let act_offset = (r>>ydec) * (width>>xdec) + (c>>xdec);
+        let temp_luma: u32 = luma[luma_offset].into();
+        let temp_mean: u32 = mean[act_offset];
+        var[act_offset] += (temp_luma as i32 - temp_mean as i32).pow(2) as u32;
+        self.average_variance += var[act_offset] as f64 / (64f64 * (width>>xdec) as f64 * (height>>ydec) as f64);
+      }
+    }
+    var.iter_mut().for_each(|sum| *sum = (((*sum >> (xdec+ydec)) as f32).sqrt()) as u32 * 2);
+
+    //Activity values are normalized to a 16-bit scale
+    let max = var.iter().max().unwrap();
+    let activity_mask: &mut [u16] = &mut *self.activity_mask.data;
+    activity_mask
+      .iter_mut()
+      .zip(var.iter())
+      .map(|(var1, var2)| *var1 = if *max!=0 {65535 * var2 / max} else {0} as u16 )
```

Comment:
Looking at your WIP patch for activity masking, I think you may actually want to skip the normalization step.

Reply:
Is there any use case for the normalized activity mask?
I have stored the mean variance of blocks in a frame in [another variable](https://github.com/xiph/rav1e/pull/1263/commits/0704c05356ff71dd0bb3a6ff4b9372adf88394be#diff-26006a0497afb96d876927e114a7b29eR303) in case I need to denormalize these values.
But if there are no use cases for a normalized mask, then I'll remove this step and the mean variance value.